contrib/win32/win32compat/w32log.c

```diff
@@ -115,16 +118,36 @@ openlog_file()
 		wcscat_s(ssh_cfg_path, _countof(ssh_cfg_path), L"\\ssh"); /* "%programData%\\ssh" */
 
 		if ((wcsncat_s(log_file, PATH_MAX + 12, ssh_cfg_path, wcslen(ssh_cfg_path)) != 0) ||
-		    (wcsncat_s(log_file, PATH_MAX + 12, logs_dir, 6) != 0) ||
-		    (wcsncat_s(log_file, PATH_MAX + 12, tail + 1, wcslen(tail + 1) - 3) != 0 ) ||
-		    (wcsncat_s(log_file, PATH_MAX + 12, L"log", 3) != 0))
+			(wcsncat_s(log_file, PATH_MAX + 12, logs_dir, 6) != 0))
 			return;
+		
+		if (strcmp(identity, "sftp-server") == 0) {
```

Comment:
rewrite this logic,

`
wchar_t *tmp_identity = NULL;
if (strcmp(identity, "sftp-server") == 0) {
    tmp_identity = utf8_to_utf16(identity);
} else{
    tmp_identity = malloc();
    tmp_identiy = <copy_from_tail>
}
`

With this logic, you can use the existing logic from lines 118-120 with a change to line 119. 
This removes the need for having separate logic